Client/core/CCore.cpp

```diff
@@ -60,13 +60,13 @@ static HMODULE WINAPI SkipDirectPlay_LoadLibraryA(LPCSTR fileName)
         const fs::path inLaunchDir = fs::path{FromUTF8(GetLaunchPath())} / "enbseries" / "enbhelper.dll";
 
         if (fs::is_regular_file(inLaunchDir, ec))
-            return Win32LoadLibraryA(inLaunchDir.u8string().c_str());
+            return Win32LoadLibraryA(inLaunchDir.string().c_str());
```

Comment:
leave it as u8string for utf8 compatibility

Reply:
Why is this conversation marked as resolved?

Reply:
See below. In C++20 path.u8string may/will return std::basic_string<char8_t> which is incompatible with std::string

Reply:
Something you're not aware of: in the MSVC STL implementation of `std::filesystem::path` the actual path is stored as a wide string (`using string_type = std::wstring`) and the `path::u8string` function uses `_Convert_wide_to_narrow` to convert the wide string to a narrow variant with the UTF-8 codepage there, but the `path::string()` function uses the active code page (ACP), which isn't going to be UTF-8 in most cases.

Reply:
> Something you're not aware of: in the MSVC STL implementation of `std::filesystem::path` the actual path is stored as a wide string (`using string_type = std::wstring`) and the `path::u8string` function uses `_Convert_wide_to_narrow` to convert the wide string to a narrow variant with the UTF-8 codepage there, but the `path::string()` function uses the active code page (ACP), which isn't going to be UTF-8 in most cases.

Should I use ToACP for u8string? If I understood correctly?

Reply:
> > Something you're not aware of: in the MSVC STL implementation of `std::filesystem::path` the actual path is stored as a wide string (`using string_type = std::wstring`) and the `path::u8string` function uses `_Convert_wide_to_narrow` to convert the wide string to a narrow variant with the UTF-8 codepage there, but the `path::string()` function uses the active code page (ACP), which isn't going to be UTF-8 in most cases.
> 
> Should I use ToACP for u8string? I understood correctly? For example: `ToACP(inLaunchDir.u8string()).c_str();`

Reply:
If you are going to call Windows API functions like `LoadLibraryA` with the `*A` suffix, then you need the active code page encoding (ACP), which you get from `path::string()`. For any other code that expects UTF-8, you have to take the wide string `path::wstring()` and convert it with `ToUTF8` (one of our functions - but please double-check that it actually converts from wide to narrow UTF-8), or you continue using `path::u8string()` but cast the values from `char8_t*` to `char*`, for example.